static/src/javascripts/projects/common/modules/crosswords/crossword.js

```diff
@@ -0,0 +1,772 @@
+                });
+            }
+        } else {
+            this.setState({
+                showAnagramHelper: false,
+            });
+        }
+    }
+
+    onClickHiddenInput(event: Event): void {
+        const focussed = this.state.cellInFocus;
+
+        this.onSelect(focussed.x, focussed.y);
+
+        /* We need to handle touch seperately as touching an input on iPhone does not fire the
+         click event - listen for a touchStart and preventDefault to avoid calling onSelect twice on
+         devices that fire click AND touch events. The click event doesn't fire only when the input is already focused */
+        if (event.type === 'touchstart') {
+            event.preventDefault();
+        }
+    }
+
+    setGridHeight(): void {
+        if (!this.$gridWrapper) {
+            this.$gridWrapper = $(React.findDOMNode(this.refs.gridWrapper));
+        }
+
+        if (
+            isBreakpoint({
+                max: 'tablet',
+            })
+        ) {
+            fastdom.read(() => {
+                // Our grid is a square, set the height of the grid wrapper
+                // to the width of the grid wrapper
+                fastdom.write(() => {
+                    this.$gridWrapper.css(
+                        'height',
+                        `${this.$gridWrapper.offset().width}px`
+                    );
+                });
+                this.gridHeightIsSet = true;
+            });
+        } else if (this.gridHeightIsSet) {
+            // Remove inline style if tablet and wider
+            this.$gridWrapper.attr('style', '');
+        }
+    }
+
+    setCellValue(x: number, y: number, value: string): void {
+        this.setState({
+            grid: helpers.mapGrid(this.state.grid, (cell, gridX, gridY) => {
+                if (gridX === x && gridY === y) {
+                    cell.value = value;
+                    cell.isError = false;
+                }
+
+                return cell;
+            }),
+        });
+    }
+
+    getCellValue(x: number, y: number): string {
+        return this.state.grid[x][y].value;
+    }
+
+    setReturnPosition(position: number): void {
+        this.returnPosition = position;
+    }
+
+    insertCharacter(character: string): void {
+        const cell = this.state.cellInFocus;
+        if (/[A-Z]/.test(character) && character.length === 1) {
+            this.setCellValue(cell.x, cell.y, character);
+            this.save();
+            this.focusNext();
+        }
+    }
+
+    cellIsEmpty(x: number, y: number): boolean {
+        return !this.getCellValue(x, y);
+    }
+
+    goToReturnPosition(): void {
+        if (
+            isBreakpoint({
+                max: 'mobile',
+            })
+        ) {
+            if (this.returnPosition) {
+                scrollTo(this.returnPosition, 250, 'easeOutQuad');
+            }
+            this.returnPosition = null;
+        }
+    }
+
+    indexOfClueInFocus(): number {
```

Comment:
Could we change this into:

```js
isClueInFocus(): number {
  return this.props.data.entries.includes(this.clueInFocus());
}
```

It's used 2 times and would simplify `focusPreviousClue`: so instead of 

https://github.com/guardian/frontend/blob/39f35f7562028c90cf6d6703d7a58b15d1aa762c/static/src/javascripts/projects/common/modules/crosswords/crossword.js#L366-L378

we could write

```js
focusPreviousClue(): void {
    const inFocus = this.indexOfClueInFocus();
    const entries = this.props.data.entries;

    if (inFocus) {
        const newClue = entries[i === 0 ? entries.length - 1 : i - 1];
        this.focusClue(
            newClue.position.x,
            newClue.position.y,
            newClue.direction
        );
    }
}
```

Reply:
Where would `i` come from in `entries[i === 0 ? entries.length - 1 : i - 1]` come from in that new version? I agree there's a bit of a conflation of concerns here (oh for an `Option`!), but I do think we need an index.

Reply:
Oh, you are right. Sorry - didn't see this 🙀